src/Microsoft.ML.AutoML/Utils/SplitUtil.cs

```diff
@@ -47,16 +47,18 @@ internal static class SplitUtil
         /// Split the data into a single train/test split.
         /// </summary>
         public static (IDataView trainData, IDataView validationData) TrainValidateSplit(MLContext context, IDataView trainData,
-            string samplingKeyColumn)
+            string samplingKeyColumn, bool shuffle = false)
```

Comment:
As a continuation of previous comments https://github.com/dotnet/machinelearning/pull/4384#pullrequestreview-307464610

---

Shuffling is helpful as part of the trainer options. No need to have a shuffle here.

```suggestion
            string samplingKeyColumn)
```

Reply:
I disagree here, I have seen accuracy go down with shuffling turned off, this has been my experience after benchmarking several image datasets on ImageClassificationTrainer. 

---
In reply to: [339329734](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339329734) [](ancestors = 339329734)

Reply:
Similar to https://github.com/dotnet/machinelearning/pull/4395#discussion_r339375063, you may be seeing noisy numbers, which makes it difficult to tell if the shuffling is impactful or not. 

For a NN, I would expect if your learning rate is high, or if you have a very low number of epochs, then shuffling can be helpful. Due to the block shuffle style in ML.NET, you won't see gains when you have above ~1k images per class.

If the new trainer prefers shuffled data, we should align with the rest of trainers which offer a hyperparameter for shuffling. 

As [noted](https://github.com/dotnet/machinelearning/pull/4384#discussion_r339248471) by @daholste, the shuffling is currently out of place:
> Perhaps conceptually 'shuffle' belongs outside of the train/valid split API ?
> If desired (and only desired for images right now), you could shuffle the results of the this API
> For instance, I don't think it's part of the ML.NET train/valid split API

Reply:
thanks, we can discuss this offline. To begin with, we should be asking the user for a trainset and a validation set, this how other APIs such as TensorFlow SLIM's image classification API is designed. Our ImageClassification API is also designed the same way but unfortunately the Model Builder UI does not accept a validation set so we are left with creating a validation set from the train set, this is troublesome in scenarios like image classification where users give a folder to all their images and within this folder you have sub folders that contain images for various classes, now if we don't shuffle we could end up training on some classes and then validating on classes that we didn;'t even train on! hence we shuffle before we create train and validation set, and FYI, Image classification API does also shuffle the train set it receives to keep data points independent. 

I can chat offline and close on this. thanks all.

Reply:
> ... now if we don't shuffle we could end up training on some classes and then validating on classes that we didn't even train on! hence we shuffle before we create train and validation set

This is an incorrect understanding of how our TrainTest and CV splitting works. More detail here: https://github.com/dotnet/machinelearning/pull/4395#discussion_r339329519

Reply:
also, can we please move shuffle outside of the train/test split API? Isn't it semantically separate from a train/test split? At the very least, since shuffle is only used in the images path, why not have a single line of code shuffling the training data before passing it in to train/test split?